src/components/bars/ProgressBar.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from 'react';
+import '../../styles/components/bars/ProgressBar.css';
+
+const ProgressBar = ({ startTime }) => {
+    const today = new Date();
+
+    const checkHour = () => {
+        const shour = parseInt(startTime.sHour);
+
+        if (startTime.status === 'FAST') {
+            if (shour + 1 >= 24) return shour - 23;
+            return shour + 1;
+        }
+        else {
+            if (shour + 14 >= 24) return shour - 10;
+            return shour + 14;
+        }
+    }
+    const checkLeftTime = () => {
+        const currentHour = today.getHours();
+        const currentMinute = today.getMinutes();
+        let endHour = endTime.eHour;
+
+        if (currentHour > endTime.eHour) {
+            endHour = endTime.eHour + 24;
+        }
+
+        const sumCurrentTime = currentHour * 60 + currentMinute;
+        const sumEndTime = endHour * 60 + endTime.eMinute;
+
+        const leftVal = sumEndTime - sumCurrentTime;
+
+        const leftHour = parseInt(leftVal / 60);
+        const leftMinute = leftVal % 60;
+
+        const standard = startTime.status === 'FAST' ? 60 : 14 * 60;
+        const tPercent = 100 - (100 * leftVal / standard);
+
+        setLeftTime({
+            lHour: leftHour,
+            lMinute: leftMinute,
+            percent: tPercent,
+            pxPercent: tPercent * 2.6
+        })
+    }
```

Comment:
You implemented the progress bar logic by using the variables appropriately.
Easy to understand variable names !

Reply:
thanks